app/src/main/java/com/omkarmoghe/pokemap/views/MainActivity.java

```diff
@@ -170,6 +170,12 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
     }
 
     private void startNotificationService(){
+
+        // check for if the service is already running
+        if (PokemonNotificationService.isRunning()) {
+            stopNotificationService();
```

Comment:
If we are starting the service, why are we turning it off then back on again? Wouldn't it just be better if we are starting the notification service, if it is running just to let it keep running the same instance? So if it wasn't running then you turn it on.